docs/docs/noir/concepts/data_types/arrays.md

```diff
@@ -255,6 +255,23 @@ fn main() {
 }
 ```
 
+### concat
+
+Concatenates this array with another array.
+
+```rust
+fn concat<let M: u32>(self, array2: [T; M]) -> [T; N + M]
+```
+
+```rust
+fn main() {
+    let arr1 = [1, 2, 3, 4];
+    let arr2 = [6, 7, 8, 9, 10, 11];
+    let concatenated_arr = arr1.concat(arr2);
+    assert(concatenated_arr == [1, 2, 3, 4, 6, 7, 8, 9, 10, 11]);
+}
```

Comment:
Can you snippet-ify this like in the aztec docs?

Reply:
I couldn't tell if you had some kind of clever code snippet stuff in the Noir repo, so I hand-wrote this. Do you have something? Or are you asking if it'd be possible to set something up like the snippet-ifying that we do in aztec?

Reply:
We pulled across the same snippet code that you added to the aztec-packages docs a while back. You can use the exact same syntax as in the aztec docs to insert snippets of noir code into the docs here.

Reply:
Perhaps that's one for a separate, dedicated PR, because I don't think any of `noir_stdlib/src/array/mod.nr` is using that code snippet tool yet.

Reply:
https://github.com/noir-lang/noir/issues/7251